cygnus-twitter/README.md

```diff
@@ -1 +1,296 @@
+    * [Hardware requirements](#section2.1)
+    * [Configuration](#section2.3)
+    * [Unit testing](#section2.5)
+    * [Management API overview](#section2.7)
+* [Features summary](#section5)
+* [Licensing](#section6)
+* [Reporting issues and contact information](#section7)
+
+##<a name="section1"></a>Welcome to Cygnus-twitter
+This project is part of [FIWARE](http://fiware.org), being part of the [Cosmos](http://catalogue.fiware.org/enablers/bigdata-analysis-cosmos) Ecosystem.
+
+Cygnus-twitter is a connector in charge of persisting tweets (https://dev.twitter.com/overview/api/tweets) in certain configured third-party storages, creating a historical view of such data.
+
+Internally, Cygnus-twitter is based on [Apache Flume](http://flume.apache.org/) and [Twitter4j](http://twitter4j.org/en/index.html). In fact, Cygnus-twitter is a Flume agent, which is basically composed of a source in charge of receiving the data from Twitter (https://twitter.com/), a channel where the source puts the data once it has been transformed into a Flume event, and a sink, which takes Flume events from the channel in order to persist the data within its body into a third-party storage.
+
+Current stable release is able to persist Twitter data in:
+
+* [HDFS](http://hadoop.apache.org/docs/current/hadoop-project-dist/hadoop-hdfs/HdfsUserGuide.html), the [Hadoop](http://hadoop.apache.org/) distributed file system.
+
+You may consider to visit [Cygnus-twitter User and Programmer Guide](doc/cygnus-twitter/user_and_programmer_guide/README.md) before going deep into the details.
+
+[Top](#top)
+
+##<a name="section2"></a>Basic operation
+###<a name="section2.1"></a>Hardware requirements
+* RAM: 1 GB, specially if abusing of the batching mechanism.
+* HDD: A few GB may be enough unless the channel types are configured as `FileChannel` type.
+
+[Top](#top)
+
+
+###<a name="section2.3"></a>Configuration
+Cygnus-twitter is a tool with a high degree of configuration required for properly running it.
+
+So, the starting point is choosing the internal architecture of the Cygnus agent. Let's assume the simplest one:
+
+```
++----------------+    +----------------+    +----------------+
+| Twitter source |----| memory channel |----|  Twitter Sink  |
++----------------+    +----------------+    +----------------+
+```
+
+Attending to the above architecture, the content of `/usr/cygnus/conf/cygnus_1.conf` will specify the following parameters:
+
+* The source:
+`cygnus-twitter.sources.http-source.type = org.telefonica.iot.cygnus.sources.TwitterSource`
+
+* The keyworks (hashtags) that are used in the twitter query to filter tweets with an specific keyword(s):
+        
+`cygnus-twitter.sources.twitter-source.keywords = keyword1, keyword2, keyword3`
+
+* The coordinates to specify the spatial area where the source will collect geo-located tweets. The coordinates will be used in the twitter query:
+
+```
+        cygnus-twitter.sources.twitter-source.south-west_latitude = 39.4247692
```

Comment:
Because these lines are eclosed by ```   ```, the 8 while spaces indentation is not necessary.